Automattic-Tracks-iOS/Event Logging/EventLoggingUploadManager.swift

```diff
@@ -0,0 +1,52 @@
+import Foundation
+import CocoaLumberjack
+import Sodium
+
+class EventLoggingUploadManager {
+
+    private struct Constants {
+        static let uuidHeaderKey = "log-uuid"
+        static let uploadHttpMethod = "POST"
+    }
+
+    internal var networkService = EventLoggingNetworkService()
+
+    internal var dataSource: EventLoggingDataSource!
+    internal var delegate: EventLoggingDelegate!
```

Comment:
I am not sure why these are `internal`, but how would you feel about getting them as constructor parameters instead? Similar to my previous comments (in terms of when I wrote them, but I know this will probably show up first) this would make it easier to test this class and also make it more customizable. We could even make these non-nullable this way (if we want) and get rid of the `precondition`s. What do you think?

Reply:
That's a better solution for sure. It'll require re-working the wrapping API for this, but this will force a better solution for ensuring they're present. I probably should've thought of this!! 🙃

Reply:
You both made a good decision here. 🙂